lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1074,9 +1074,13 @@ def _check_certs_expiration(self, event: UpdateStatusEvent) -> None:
 
     def _scrape_config(self) -> List[Dict]:
         """Generates the scrape config as needed."""
-        app_secrets = self.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val)
-        ca = app_secrets.get("ca-cert")
-        pwd = self.secrets.get(Scope.APP, self.secrets.password_key(COSUser))
+        if (
+            not (app_secrets := self.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val))
+            or not (ca := app_secrets.get("ca-cert"))
+            or not (pwd := self.secrets.get(Scope.APP, self.secrets.password_key(COSUser)))
+        ):
+            # Not yet ready, waiting for certain values to be set
+            return []
```

Comment:
When is the right config set again?

Reply:
@Mehdi-Bendriss this is related to the COS user setup. @juditnovak may correct me here but we need the start process to be over, as the user is added at post-start-init. By then, we should also receive the CA certs.

Reply:
Yes - so my question is: once the CA is set, and COSUser created, how does the `self.cos_integration.scrape_configs` object get updated with these 2 new values (knowing it was already set to `[]`)?

Reply:
@Mehdi-Bendriss that method is called every time the charm comes up, [here](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_base_charm.py#L148). So, it starts with "empty" but eventually, we can populate this method.

Btw, that is the standard way charms deal with COS provider, by calling them at the constructor of the main charm.